src/pt/mangalivre/src/eu/kanade/tachiyomi/extension/pt/mangalivre/MangaLivreImageExtractor.kt

```diff
@@ -9,43 +9,51 @@ import org.jsoup.nodes.Document
+    private const val DEFAULT_KEY2 = "29d076b6"
 
     fun extractImageUrls(document: Document, json: Json): List<String>? {
-        for (script in document.select("script:not([src])")) {
-            val data = script.data()
-            if (data.length < 100) continue
+        val dataXSpans = document.select("span[data-x]")
+        if (dataXSpans.isEmpty()) return null
 
-            if (data.contains("idx++") && data.contains("var _x = [")) {
-                extractFromXorEncryption(data, json)?.let { return it }
-            }
-        }
-        return null
-    }
+        val (key1, key2) = findXorKeysFromScript(document) ?: Pair(DEFAULT_KEY1, DEFAULT_KEY2)
 
-    private fun extractFromXorEncryption(scriptData: String, json: Json): List<String>? {
-        val dataMatches = XOR_DATA_REGEX.findAll(scriptData).toList()
-        if (dataMatches.isEmpty()) return null
+        val allEncodedData = dataXSpans.mapNotNull { span ->
+            span.attr("data-x").takeIf { it.isNotBlank() }
+        }.joinToString("")
 
-        val joinedBase64 = dataMatches.joinToString("") { it.groupValues[1] }
-        if (joinedBase64.isEmpty()) return null
+        if (allEncodedData.isBlank()) return null
 
-        val xorKey = XOR_KEY_REGEX.findAll(scriptData)
-            .map { it.groupValues[1] }
-            .find { it.length == 8 }
-            ?: return null
+        val jsonArray = decodeDataXToJson(allEncodedData, key1, key2)
+        return parseUrlsFromJson(jsonArray, json)
+    }
 
-        return try {
-            val decodedBytes = Base64.decode(joinedBase64, Base64.DEFAULT)
+    private fun decodeDataXToJson(encodedData: String, key1: String, key2: String): String {
+        val decodedBytes = Base64.decode(encodedData, Base64.DEFAULT)
 
-            val decrypted = buildString {
-                for (i in decodedBytes.indices) {
-                    append((decodedBytes[i].toInt() xor xorKey[i % xorKey.length].code).toChar())
-                }
+        val result = buildString {
+            for (i in decodedBytes.indices) {
+                val block = i / 8
+                val key = if (block % 2 == 0) key1 else key2
+                val c = decodedBytes[i].toInt() and 0xFF xor key[i % 8].code
+                append(c.toChar())
             }
+        }
+
+        val startIdx = result.indexOf('[')
+        val endIdx = result.lastIndexOf(']')
+        if (startIdx == -1 || endIdx == -1 || endIdx <= startIdx) {
+            return result
+        }
 
-            json.parseToJsonElement(decrypted).jsonArray
+        return result.substring(startIdx, endIdx + 1)
+            .replace("\\/", "/")
+    }
+
+    private fun parseUrlsFromJson(jsonStr: String, json: Json): List<String>? {
+        return try {
+            json.parseToJsonElement(jsonStr).jsonArray
                 .map { it.jsonPrimitive.content.trim() }
                 .filter { it.isNotBlank() && isValidImageUrl(it) }
```

Comment:
```suggestion
                .filter { isValidImageUrl(it) }
```

`isValidImageUrl` already ensures non-blankness.